File: wisdom-education-admin-front/src/enum/QuestionTypeEnum.js
/**
 * 试题类型枚举
 * @type {
 * {CALCULATION_QUESTION: {code: number, value: string},
 * SYNTHESIS_QUESTION: {code: number, value: string},
 * FILL_QUESTION: {code: number, value: string},
 * JUDGMENT_QUESTION: {code: number, value: string},
 * MULTIPLE_QUESTION: {code: number, value: string},
 * SINGLE_QUESTION: {code: number, value: string},
 * INDEFINITE_ITEM_QUESTION: {code: number, value: string}}}
 */
const QuestionType = {
  SINGLE_QUESTION: {
    'code': 1,
    'value': '单选题'
  },

  MULTIPLE_QUESTION: {
    'code': 2,
    'value': '多选题'
  },

  FILL_QUESTION: {
    'code': 3,
    'value': '填空题'
  },

  SYNTHESIS_QUESTION: {
    'code': 4,
    'value': '综合题'
  },

  INDEFINITE_ITEM_QUESTION: {
    'code': 5,
    'value': '不定项题'
  },

  JUDGMENT_QUESTION: {
    'code': 6,
    'value': '判断题'
  },

  CALCULATION_QUESTION: {
    'code': 7,
    'value': '计算题'
  }
}

/**
 * 根据code获取试题类型名称
 * @param code
 * @returns {*}
 */
export function getByCode (code) {
  let values = Object.values(QuestionType)
  for (let i = 0; i < values.length; i++) {
    if (code === values[i].code) {
      return values[i].value
    }
  }
}

export default QuestionType

File: wisdom-education/education-common/src/main/java/com/education/common/enums/QuestionTypeEnum.java
package com.education.common.enums;

public enum QuestionTypeEnum {
    SINGLE_QUESTION(1, "单选题"), MULTIPLE_QUESTION(2, "多选题"),
    FILL_QUESTION(3, "填空题"), SYNTHESIS_QUESTION(4, "综合题"),
    INDEFINITE_ITEM_QUESTION(5, "不定项题"),
    CALCULATION_QUESTION(7, "计算题"),
    JUDGMENT_QUESTION(6, "判断题");
    private Integer value;
    private String name;
    QuestionTypeEnum(Integer value, String name) {
        this.name = name;
        this.value = value;
    }

    public Integer getValue() {
        return value;
    }

    public String getName() {
        return name;
    }

    public static String getName(Integer type) {
        QuestionTypeEnum questionTypes[] = values();
        for (QuestionTypeEnum questionType : questionTypes) {
            if (questionType.getValue() == type) {
                return questionType.getName();
            }
        }
        return null;
    }

    /***
     * 是否为客观题
     * @param questionType
     * @return
     */
    public static boolean isObjectiveQuestion(Integer questionType) {
        if (SINGLE_QUESTION.getValue().equals(questionType)
                || MULTIPLE_QUESTION.getValue().equals(questionType)
                || JUDGMENT_QUESTION.getValue().equals(questionType)) {
            return true;
        }
        return false;
    }
}